src/Components/test/E2ETest/Infrastructure/BasicTestAppTestBase.cs

```diff
@@ -38,15 +41,38 @@ public string ServerPathBase
         protected SelectElement WaitUntilTestSelectorReady()
         {
             var elemToFind = By.CssSelector("#test-selector > select");
-            WaitUntilExists(elemToFind, timeoutSeconds: 30);
+            WaitUntilExists(elemToFind, timeoutSeconds: 30, throwOnError: true);
```

Comment:
Right now I've only wired this up here because the goal is to capture failures in the initial load. 

I think it would also be interesting to apply the same thing to where we select a "test component" with the principle that we should write tests that throw an exception during the initial load the UI. It would be OK to throw an unhandled exception as the result of a button press.

One thing I don't know is whether this will cause problems if we do have tests that throw due to ordering.

It seems reasonable that whatever we do to "reset" state between tests would also want to clear the logs, and that would help.

Reply:
We reuse the same browser instance for all tests on a given class, so this can cause issues in that scenario, if a test fails, then the rest will start failing, although that might be ok as they would likely run into the same issue.

Could we have a set of sentinel tests that simply load the app and validate that there are no errors? You can run those tests first to ensure that the scripts compile and do the right things.

Reply:
> We reuse the same browser instance for all tests on a given class, so this can cause issues in that scenario, if a test fails, then the rest will start failing, although that might be ok as they would likely run into the same issue.

right, that's a valid concern. I'm not giving up on this idea just yet though. We could probably clear the log between tests to address that.

Reply:
I'm not sure you can clear the log through selenium, but give it a try.

Reply:
This is goodness, BTW, we just need to find the right spot.

Reply:
Looks like you can call `console.clear()` (e.g., using Selenium's `IJavaScriptExecutor` which we also use elsewhere).

Reply:
All the tests could start by doing that. We could put this call in `BrowserTestBase`'s `InitializeAsync`.